src/panels/lovelace/components/hui-entity-editor.ts

```diff
@@ -1,33 +1,60 @@
-import "../../../components/ha-icon-button";
+import { mdiDrag } from "@mdi/js";
 import {
   css,
   CSSResult,
   customElement,
   html,
+  internalProperty,
   LitElement,
   property,
+  PropertyValues,
   TemplateResult,
 } from "lit-element";
+import { guard } from "lit-html/directives/guard";
+import { SortableEvent } from "sortablejs";
+import Sortable, {
+  AutoScroll,
+  OnSpill,
+} from "sortablejs/modular/sortable.core.esm";
 import { fireEvent } from "../../../common/dom/fire_event";
 import "../../../components/entity/ha-entity-picker";
+import "../../../components/ha-icon-button";
+import { sortStyles } from "../../../components/ha-sidebar-sort-styles";
 import { HomeAssistant } from "../../../types";
 import { EditorTarget } from "../editor/types";
 import { EntityConfig } from "../entity-rows/types";
 
 @customElement("hui-entity-editor")
 export class HuiEntityEditor extends LitElement {
-  @property() protected hass?: HomeAssistant;
+  @property({ attribute: false }) protected hass?: HomeAssistant;
 
-  @property() protected entities?: EntityConfig[];
+  @property({ attribute: false }) protected entities?: EntityConfig[];
 
   @property() protected label?: string;
 
+  @internalProperty() private _attached = false;
+
+  @internalProperty() private _renderEmptySortable = false;
+
+  private _sortable?;
+
+  public connectedCallback() {
+    super.connectedCallback();
+    this._attached = true;
+  }
+
+  public disconnectedCallback() {
+    super.disconnectedCallback();
+    this._attached = false;
+  }
+
   protected render(): TemplateResult {
     if (!this.entities) {
       return html``;
     }
 
     return html`
+      ${sortStyles}
```

Comment:
Put this in `styles()`

Reply:
ITs a template result

Reply:
So change it to CSSResult :-)

Reply:
Should rename it anyway when we are going to use it on multiple places

Reply:
Do you know how to dynamically return styles in `static get styles()` ? So that we only append if `this.editMode` in `HaSidebar`?

Reply:
use
```
<style>
dragStyles.cssText
</style>
```